bigquery/snippets/snippet.go

```diff
@@ -25,6 +26,49 @@ func createDataset(client *bigquery.Client, datasetID string) error {
 	return nil
 }
 
+func updateDatasetDescription(client *bigquery.Client, datasetID string) error {
```

Comment:
Looks like `updateDatasetDescription` checks `ds.Metadata`, but `updateDatasetDefaultExpiration` and `deleteEmptyDataset` don't. Is that intentional? I don't quite understand `original.ETag` and when to use it, either.

Reply:
It was intentional.  Per offline discussion with tswast, samples will leverage use of the ETag for safe updates.